hepstats/hypotests/calculators/basecalculator.py

```diff
@@ -45,149 +46,15 @@ def __init__(self, input, minimizer):
                 >>> calc = BaseCalculator(input=loss, minimizer=MinuitMinimizer())
         """
 
-        if is_valid_fitresult(input):
-            self._loss = input.loss
-            self._bestfit = input
-        elif is_valid_loss(input):
-            self._loss = input
-            self._bestfit = None
-        else:
-            raise ValueError("{} is not a valid loss funtion or fit result!".format(input))
-
-        if not is_valid_minimizer(minimizer):
-            raise ValueError("{} is not a valid minimizer !".format(minimizer))
+        super(BaseCalculator, self).__init__(input, minimizer)
 
-        self._minimizer = minimizer
-        self.minimizer.verbosity = 0
-        # cache of the observed nll values
         self._obs_nll = {}
```

Comment:
This class now has very little more compared to its base class `HypotestsObject`. It's not easy to see what extra the class provides wrt the base class, and that's something to say in the docstring I reckon. Otherwise, it is also not easy to see where members are defined/calculated.

The for loops below seem to also be in the base class, no? So this should be remove too, unless I'm overlooking something.

Reply:
> This class now has very little more compared to its base class HypotestsObject

What do mean by `little`? the `BaseCalculator` class defines all the methods to profile the likelihood and compute pvalues.

> The for loops below seem to also be in the base class, no? So this should be remove too, unless I'm overlooking something.

Yep this will be removed.

Reply:
There is hardly anything extra in this class compared to what is defined in the base class. That makes me wonder whether there is really need to have the separation. But I'm probably missing something, no worries ;-).

Reply:
> That makes me wonder whether there is really need to have the separation.

This class is not only used for `BaseCalculator` but also for `ToysManager`.

Reply:
OK OK.